core/src/subgraph/runner.rs

```diff
@@ -361,56 +361,49 @@ where
 
         let first_error = deterministic_errors.first().cloned();
 
-        match store.transact_block_operations(
-            block_ptr,
-            firehose_cursor,
-            mods,
-            stopwatch,
-            data_sources,
-            deterministic_errors,
-        ) {
-            Ok(_) => {
-                // For subgraphs with `nonFatalErrors` feature disabled, we consider
-                // any error as fatal.
-                //
-                // So we do an early return to make the subgraph stop processing blocks.
-                //
-                // In this scenario the only entity that is stored/transacted is the PoI,
-                // all of the others are discarded.
-                if has_errors && !is_non_fatal_errors_active {
-                    // Only the first error is reported.
-                    return Err(BlockProcessingError::Deterministic(first_error.unwrap()));
-                }
+        store
+            .transact_block_operations(
+                block_ptr,
+                firehose_cursor,
+                mods,
+                stopwatch,
+                data_sources,
+                deterministic_errors,
+            )
+            .context("Failed to transact block operations")?;
```

Comment:
Will this create multiple lines like @lutter mentioned the other day? 🤔

Reply:
Only if printed as `{:?}`, and the places that print this use `{:#}` and actually explicitly remove any newlines that sneak in due to wasm stack traces https://github.com/graphprotocol/graph-node/blob/102a96ba38393a1bc7ad3403acd7698e9bc07fba/core/src/subgraph/runner.rs#L719